Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm

```diff
@@ -238,6 +244,16 @@
             break;
 
         auto& active = *maybeActive;
+        if (WTF::atomicLoad(&active.terminateRequested))
```

Comment:
Unsolicited feedback :) 
Above, you already switch on a shared variable completely unsynchronized. It's a bit convoluted.
What's the point of ensuring reading `active.terminateRequested` correct, if the whole premise is relying on getting the read for the get_if correct? And if the Active state wasn't correct, how could anything else be. Your synchronization state, semaphores etc. are in the m_state, and the way code is written it is as if it can be destroyed at any given moment.

It would be simpler to understand if you didn't access the "Active/Idle" as a variant in the render loop.
In fact, the ARKiitCoordinatior likely shouldn't be accessed in the render loop.

Something like

```
void arKitCoordinatorRenderLoop(Ref<RenderParameters> params) {
    while(!params->shouldStop.load()) {
     ....
    }
}
```

And whether the RenderParameters  is refcounted or just plain & depends on the mode if upon stopping you want to wait until the renderloop has really ended or not.
(RenderParameters is a poor name)